modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -553,6 +555,98 @@ void cv::gimpl::GStreamingExecutor::setSource(GRunArgs &&ins)
         util::throw_error(std::logic_error("Only one video source is"
                                            " currently supported!"));
     }
+ 
+    // testing meta data is cv::GMatDesc{CV_8U,3,cv::Size{768,576}}
+    if (m_metas.empty()) {
+        // extract meta from first frame
+    } else {
+        //const auto& metas = m_metas;
+    }
+    
+    const auto& metas = m_metas;
+    
+    if (/*wasFinished*/true) {
+        auto pass_ctx = ade::passes::PassContext{*m_orig_graph.get()};
+        cv::gimpl::passes::initMeta(pass_ctx, metas);
+        //does inferMeta needed?? 
```

Comment:
Yes it is needed. How do you propagate the metadata information in the graph otherwise?